lib/msf/core/exploit/remote/http/kubernetes/client.rb

```diff
@@ -273,6 +273,8 @@ def call_api(request, options = {})
                 raise Kubernetes::Error::ForbiddenError.new(res: res)
               elsif res.code == 404
                 raise Kubernetes::Error::NotFoundError.new(res: res)
+              elsif res.code >= 500 && res.code <= 599
+                raise Kubernetes::Error::ServerError.new(res: res)
```

Comment:
Handles the scenario where I was getting server side timeouts, and the `pod` json response was ending as:

```
[4] pry(#<Msf::Modules::Exploit__Multi__Kubernetes__Exec::MetasploitModule>)> pod
=> {:metadata=>{}, :status=>"Failure", :message=>"Timeout: request did not complete within the allotted timeout", :reason=>"Timeout", :details=>{}, :code=>504}`
```